crud/users.py

```diff
@@ -175,30 +163,82 @@ async def get_all_existing_users(user_role, db: AsyncSession):
 
 
 @handle_errors
-async def set_new_role(uid, new_role, user_role, db: AsyncSession):
+async def crud_set_role(user_id, new_role, requester_role, db: AsyncSession):
     """
     Function to change the role of an existing user (only for admin users).
     
     Returns:
-        Status code and message of the transaction.
+        Status and message of the transaction.
     """
-    query = sa.select(User).where(User.id == uid)
+    query = sa.select(User).where(User.id == user_id)
     result = await db.execute(query)
     user = result.scalar()
     if user is None:
-        raise HTTPException(status_code=400,
-                            detail=f'User with ID {uid} does not exist.')
-    if user_role != 'admin':
-        raise HTTPException(status_code=403,
-                            detail=NOT_AUTHORIZED)
+        raise_helper(404, "User", user_id)
+    if requester_role != 'admin':
+        raise_helper(401)
     if user.role == new_role.role:
-        raise HTTPException(status_code=200,
-                            detail=f'User already had role {new_role.role}. No changes made.')
+        raise_helper(422, "User", new_role)
 
     set_role = (
-        sa.update(User).where(User.id == uid).values(role=new_role.role)
+        sa.update(User).where(User.id == user_id).values(role=new_role.role)
     )
     await db.execute(set_role)
     await db.commit()
     return {'status': 'success',
-            'message': f'User {uid} successfully changed to {new_role.role}.'}
+            'message': f'User {user_id} successfully changed to {new_role.role}.'}
+
+
+@handle_errors
+async def set_reminder(user_id, reminder_value, db: AsyncSession):
+    """
+    Function to set the reminders on or off for the user.
+
+    Returns:
+        Message of the transaction.
+    """
+    query = sa.select(User).where(User.id == user_id)
+    result = await db.execute(query)
+    user = result.scalar()
+
+    if user.daily_reminder == reminder_value:
+        status = "enabled" if reminder_value else "disabled"
+        raise_helper(422, "Reminders", status)
+
+    reminders_config = (
+        sa.update(User).where(User.id == user_id).values(daily_reminder=reminder_value)
+    )
+    await db.execute(reminders_config)
+    await db.commit()
+    return {'status': 'success', 'message': 'Reminders successfully configured.'}
+
+
+@handle_errors
+async def crud_send_daily_reminder(timezone, user_role, db: AsyncSession):
+    """
+    Function to send the reminders to users with this function
+
+    Returns:
+        Message of the transaction.
+    """
+    if user_role != 'admin':
+        raise_helper(401)
+    tasks_url = app_url + ":8080/api/v1/tasks/"
+    query = sa.select(User.email, func.string_agg(
+        'Title: ' + Todo.title + '\nDescription: ' + Todo.description +
```

Comment:
I see thats thats you start using templates. Good job
Why wont you render Titel, Description ... via template itself?
Just select data via sql and generate text via template.
Also as I wrote before, why not move rendering logic to separate function?